src/components/PDFView/index.js

```diff
@@ -1,54 +1,31 @@
 import 'core-js/features/array/at';
-import pdfWorkerSource from 'pdfjs-dist/legacy/build/pdf.worker';
 import React, {Component} from 'react';
+import {PDFPreviewer} from 'react-fast-pdf';
 import {View} from 'react-native';
 import {withOnyx} from 'react-native-onyx';
-import {pdfjs} from 'react-pdf';
 import _ from 'underscore';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
 import PressableWithoutFeedback from '@components/Pressable/PressableWithoutFeedback';
+import Text from '@components/Text';
 import withLocalize from '@components/withLocalize';
 import withThemeStyles from '@components/withThemeStyles';
 import withWindowDimensions from '@components/withWindowDimensions';
 import compose from '@libs/compose';
-import Log from '@libs/Log';
 import variables from '@styles/variables';
 import * as CanvasSize from '@userActions/CanvasSize';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import PDFViewConstants from './constants';
 import PDFPasswordForm from './PDFPasswordForm';
 import * as pdfViewPropTypes from './pdfViewPropTypes';
-import PDFDocument from './WebPDFDocument';
 
 class PDFView extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            numPages: null,
-            pageViewports: [],
-            containerWidth: props.windowWidth,
-            containerHeight: props.windowHeight,
-            password: undefined,
-            /** used to keep the PDFPasswordForm mounted (for it to maintain state) while password is being verified */
-            isCheckingPassword: false,
-            isPasswordInvalid: false,
             isKeyboardOpen: false,
         };
-        this.onDocumentLoadSuccess = this.onDocumentLoadSuccess.bind(this);
-        this.initiatePasswordChallenge = this.initiatePasswordChallenge.bind(this);
-        this.attemptPDFLoad = this.attemptPDFLoad.bind(this);
         this.toggleKeyboardOnSmallScreens = this.toggleKeyboardOnSmallScreens.bind(this);
-        this.calculatePageHeight = this.calculatePageHeight.bind(this);
-        this.calculatePageWidth = this.calculatePageWidth.bind(this);
-        this.getDevicePixelRatio = _.memoize(this.getDevicePixelRatio.bind(this));
-        this.setListAttributes = this.setListAttributes.bind(this);
-
-        this.documentOpenedSuccessfully = false;
-
-        const workerURL = URL.createObjectURL(new Blob([pdfWorkerSource], {type: 'text/javascript'}));
-        if (pdfjs.GlobalWorkerOptions.workerSrc !== workerURL) {
-            pdfjs.GlobalWorkerOptions.workerSrc = workerURL;
-        }
```

Comment:
Regarding the part replacing this code. Please make sure it works offline, this part https://github.com/Expensify/react-fast-pdf/blob/12cde1abe62cd7ba162a185f020597f149f49da4/src/PDFPreviewer.tsx#L61 does not seem to work well when offline.

(Bug found while working on another PR https://github.com/Expensify/App/pull/35255#discussion_r1506951211)

Reply:
Thank you for info. Will check it.

Reply:
Can we create a separate issue for this task? Looks like it is a hard one. Moreover, it is a legacy bug. As I said previously, this PR does not introduce any new UI/UX improvements or bug-fixes. cc @allroundexperts @s77rt

Reply:
Btw, I am improving `react-fast-pdf` to work smoothly when offline. Going to update the package.

Reply:
Updated to the latest `react-fast-pdf` version with the offline fix - f9a5cce.